### PR TITLE
Graduate `PreferNominatedNode` to GA

### LINKED ISSUE #93013: Honor "NominatedNodeName" during the new scheduling cycle
**What would you like to be added**:
A option of scheduler used to check the feasibility of `NominatedNodeName` directly instead of loop all the nodes in the cluster after pod preemption is done previously.

**Why is this needed**:
When the preemption is happened a best candidate is chosen from current nodes in the cluster, i.e. the feasibility and prioritization has been thoroughly considered. 

The best candidate is patched to be the pod's field of `NominatedNodeName`, after the new scheduling cycle is started, the "NominatedNodeName" **mostly** like is still the best candidate for the preemptor pod, but in the new scheduling cycle, all nodes are added to check the feasibility again.

Given in the a large cluster with lots of nodes, how it does like this make the efforts unnecessary heaven, we could add a new option so that we can validate the `NominatedNodeName` directly instead of looping all the nodes, and fall back to check other nodes if the `NominatedNodeName` can not pass all the filters again.

pros:
Save scheduling from running all the filter plugins for all the nodes twice.

cons:
There might be other nodes are more suitable for scheduling the pod than the `NominatedNodeName` node, if the are some big changes happened to that **nodes** after the new scheduling cycle but before the `NominatedNodeName` is re-examined.

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/93013

#### Does this PR introduce a user-facing change?

```release-note
Feature of `PreferNominatedNode` is graduated  to GA
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/pull/2026, https://github.com/kubernetes/enhancements/pull/3057
```

### CODE REVIEW

File: test/integration/scheduler/preemption_test.go [hunk 1]
```diff
@@ -1307,137 +1307,3 @@ func TestPDBInPreemption(t *testing.T) {
...
-	return testCtx
-}
-
-// TestPreferNominatedNode test when the feature of "PreferNominatedNode" is enabled, the overall scheduling logic is not changed.
-// If the nominated node pass all the filters, then preemptor pod will run on the nominated node, otherwise, it will be scheduled
-// to another node in the cluster that ables to pass all the filters.
-// NOTE: This integration test is not intending to check the logic of preemption, but rather a sanity check when the feature is
-// enabled.
-func TestPreferNominatedNode(t *testing.T) {
-	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PreferNominatedNode, true)()
```
Comment (alculquicondor) [inline test/integration/scheduler/preemption_test.go:1331]: why remove this test?

Comment (chendave) [reply test/integration/scheduler/preemption_test.go:1331]: with the feature gate removed, I was thinking this general case should be covered somewhere already. I went through all those test in the file, seems this test is an additional one, will add it back.

**Comment by alculquicondor** [comment]
/lgtm

Will approve when the KEP gets approved

**Comment by chendave** [comment]
@alculquicondor might you approve this one? thanks

**Comment by chendave** [comment]
> Will approve when the KEP gets approved

KEP is merged by the way.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

Thanks for following through this feature's graduation!

**Comment by alculquicondor** [comment]
/assign @ahg-g 

for pkg/features

**Comment by alculquicondor** [comment]
Please don't forget to update the KEP issue https://github.com/kubernetes/enhancements/issues/1923, the KEP itself https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/1923-prefer-nominated-node and website documentation

**Comment by chendave** [comment]
sure, will follow, thanks for the reminder! @alculquicondor

**Pull request merged**